packages/core/src/httpClient/basicHttpClient.ts

```diff
@@ -218,11 +223,40 @@ export class BasicHttpClient {
     return this.postRequest(rawResponse, request, mutatedRequest);
   }
 
-  private constructUrl(path: string, params: HttpQueryParams = {}) {
+  private constructUrl<T extends object>(path: string, params?: T) {
     let url = path;
-    const hasQueryParams = Object.keys(params).length > 0;
+    const hasQueryParams = !!params && Object.keys(params).length > 0;
     if (hasQueryParams) {
-      const search = new URLSearchParams(params).toString();
+      const normalizedParams: Record<string, string> = Object.entries(
+        params,
+      ).reduce(
+        (acc, [key, value]) => {
+          switch (typeof value) {
+            case 'undefined': {
+              return acc;
+            }
+            case 'string':
+            case 'number':
+            case 'boolean': {
+              acc[key] = `${value}`;
+              return acc;
+            }
+            case 'object': {
+              if (Array.isArray(value)) {
+                acc[key] = `[${value.join(',')}]`;
+              }
+              return acc;
+            }
+            default: {
+              throw new Error(
+                `Unsupported value query parameter type: ${typeof value}, ${key}: ${value}`,
+              );
+            }
+          }
+        },
+        {} as Record<string, string>,
+      );
```

Comment:
This code is well-tested (unit tests). The reason to change it was that the linter was not happy as some input to `URLSearchParams` wasn't not valid